File: src/Presentation/WebApp/Models/Dtos/AuditableDto.cs
namespace GenogramSystem.WebApp.Models.Dtos;

public class EntityDto
{
    public long             Id              { get; set; }
    public Guid             UniqueId        { get; set; }
}

public class CreatableEntityDto : EntityDto
{
    public long?            CreatedById     { get; set; }
    public DateTimeOffset?  CreatedOn       { get; set; }
    public UserDto?         CreatedBy       { get; set; }
}


public class AuditableDto : CreatableEntityDto
{
    public long?            DeletedById     { get; set; }
    public DateTimeOffset?  DeletedOn       { get; set; }
    public long?            UpdatedById     { get; set; }
    public DateTimeOffset?  UpdatedOn       { get; set; }

    public UserDto?         UpdatedBy       { get; set; }

}

File: src/Presentation/WebApp/ClientApp/src/Components/Core/Dto/AuditableDto.ts
import { IUserDto } from "./Users"

export interface IAuditableDto {
    id?: number
    uniqueId?: string
    createdById?: string
    createdOn?: Date
    deletedById?: string
    deletedOn?: Date
    updatedById?: string
    updatedOn?: Date,
    createdBy?:IUserDto
    updatedBy?:IUserDto
}